### PR TITLE
[feat]: [CDS-80860]: In trigger functor return payload payload instead of throwing error

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53772)

### CODE REVIEW

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 4 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53772)* status: 7 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/TriggerFunctor.java` line 81 at r1](https://reviewable.io/reviews/harness/harness-core/53772#-NfzaRWQ2gshcMZB5MaS:-NfzaRWQ2gshcMZB5MaT:b98089r) ([raw file](https://github.com/harness/harness-core/blob/670da4d4b18a74f1b9675f3f929e40bfc161b8a4/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/TriggerFunctor.java#L81)):*
> ```Java
>           jsonObject.put(PAYLOAD, JsonPipelineUtils.read(metadata.getTriggerJsonPayload(), List.class));
>         } catch (IOException toListEx) {
>           if (pmsFeatureFlagHelper.isEnabled(
> ```

@rathodmeetsatish will you be adding to all trigger functors in this PR?

**Review by archit-harness — APPROVED** [review]
> Reviewed 10 of 10 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53772)* status: 10 file(s) to review. 1 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/TriggerFunctor.java` line 29 at r2](https://reviewable.io/reviews/harness/harness-core/53772#-Ng4JZ2uDHMsiroUqESS:-Ng4JZ2uDHMsiroUqEST:bwas0ae) ([raw file](https://github.com/harness/harness-core/blob/dd93296a6b1cf7a9e8a066687d6bc1187bea8c36/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/TriggerFunctor.java#L29)):*
> ```Java
> import io.harness.ngtriggers.helpers.TriggerHelper;
> import io.harness.pms.contracts.ambiance.Ambiance;
> import io.harness.utils.PmsFeatureFlagHelper;
> ```

nit: this import should be deleted

**Comment by rathodmeetsatish** [comment]
trigger uts

**Review by archit-harness — DISMISSED** [review]
> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53772)* status: 10 file(s) to review. 1 of 2 approvals obtained. 

___
*[`pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/TriggerFunctor.java` line 29 at r2](https://reviewable.io/reviews/harness/harness-core/53772#-Ng4JZ2uDHMsiroUqESS:-NgJH6oO31mlqDfGS1A0:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/dd93296a6b1cf7a9e8a066687d6bc1187bea8c36/pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/TriggerFunctor.java#L29)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

nit: this import should be deleted
</blockquote></details>

done

**Review by vcalasansh — APPROVED** [review]
:lgtm:

> Reviewed 5 of 10 files at r2, 4 of 4 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53772#-:-NgJcZ2YCLWvqASKFLYu:bnfp4nl)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Comment by rathodmeetsatish** [comment]
trigger uts

**Comment by rathodmeetsatish** [comment]
trigger review

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/EventPayloadFunctor.java [hunk 1]
```diff
@@ -33,18 +33,22 @@ public EventPayloadFunctor(Ambiance ambiance, PlanExecutionMetadataService planE
 
   @Override
   public Object bind() {
+    PlanExecutionMetadata planExecutionMetadata =
+        planExecutionMetadataService.findByPlanExecutionId(ambiance.getPlanExecutionId())
+            .orElseThrow(()
```
Comment (sribalij) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/expressions/functors/EventPayloadFunctor.java:38]: Why is it IllegalStateException?

**Pull request merged**